cupy/linalg/_solve.py

```diff
@@ -223,6 +225,18 @@ def lstsq(a, b, rcond=1e-15):
 
     .. seealso:: :func:`numpy.linalg.lstsq`
     """
+    if rcond == 'warn':
+        warnings.warn(
+            '`rcond` parameter will change to the default of '
+            'machine precision times ``max(M, N)`` where M and N '
+            'are the input matrix dimensions.\n'
+            'To use the future default and silence this warning '
+            'we advise to pass `rcond=None`.\n'
+            'Using the old CuPy default: `rcond=1e-15`, which is '
+            'neither the same as the old NumPy default (`rcond=-1`).',
+            FutureWarning)
+        rcond = 1e-15
```

Comment:
The different default value is considered as a bug.
```suggestion
        rcond = -1
```

Then, the function should also implement the following spec of `*gelss`:

> If RCOND < 0, machine precision is used instead.

Reply:
It seems a reference implementation uses `if rcond >= 0 or rcond <= 1`.  I found the following code in `dlalsd`.
```fortran
       IF( ( rcond.LE.zero ) .OR. ( rcond.GE.one ) ) THEN
          rcnd = eps
       ELSE
          rcnd = rcond
       END IF
```